src/store/payment/actions.ts

```diff
@@ -23,28 +23,37 @@ import {
 } from '@src/store/payment/mutationTypes';
 import { Validity } from '@src/view_models/Validity';
 import { isValidAmount } from '@src/store/amountValidator';
+import { PaymentInitialisationPayload } from '@src/view_models/PaymentInitialisationPayload';
+import { isValidPaymentType } from '@src/store/paymentTypeValidator';
+import { isValidInterval } from '@src/store/intervalValidator';
 
 export const actions = {
 	[ discardInitialization ]( context: ActionContext<DonationPayment, any> ): void {
 		context.commit( SET_INITIALIZED, false );
 	},
-	[ initializePayment ]( context: ActionContext<DonationPayment, any>, initialValues: InitialPaymentValues ): Promise<boolean> {
-		let amountIsFilledAndValid = false, paymentIsFilled = false;
-		if ( initialValues.amount && initialValues.amount !== '0' ) {
-			amountIsFilledAndValid = isValidAmount( Number( initialValues.amount ) );
+	[ initializePayment ]( context: ActionContext<DonationPayment, any>, payload: PaymentInitialisationPayload ): Promise<boolean> {
+		const { initialValues, allowedPaymentTypes, allowedIntervals } = payload;
+
+		const amountIsValid = isValidAmount( Number( initialValues.amount ) );
+		const intervalIsValid = isValidInterval( initialValues.paymentIntervalInMonths, allowedIntervals );
+		const paymentTypeIsValid = isValidPaymentType( initialValues.type, initialValues.paymentIntervalInMonths, allowedPaymentTypes );
+
+		if ( amountIsValid ) {
 			context.commit( SET_AMOUNT, initialValues.amount );
-			context.commit( SET_AMOUNT_VALIDITY, amountIsFilledAndValid ? Validity.VALID : Validity.INVALID );
+			context.commit( SET_AMOUNT_VALIDITY, Validity.VALID );
 		}
 
-		if ( initialValues.type && initialValues.type !== '' ) {
+		context.commit( SET_INTERVAL, intervalIsValid ? initialValues.paymentIntervalInMonths : '0' );
+
+		if ( paymentTypeIsValid ) {
 			context.commit( SET_TYPE, initialValues.type );
 			context.commit( SET_TYPE_VALIDITY, Validity.VALID );
-			paymentIsFilled = true;
 		}
-		context.commit( SET_INTERVAL, initialValues.paymentIntervalInMonths );
-		context.commit( SET_INITIALIZED, amountIsFilledAndValid && paymentIsFilled );
 
-		return Promise.resolve( amountIsFilledAndValid && paymentIsFilled );
+		const paymentIsFilledAndValid = amountIsValid && paymentTypeIsValid;
```

Comment:
```suggestion
		const paymentIsFilledAndValid = amountIsValid && paymentTypeIsValid && intervalIsValid;
```

I'd say to stop on the 1st page even when the interval could be "rescued" by setting it to one-time, but that is for the PM to decide